src/Sulu/Bundle/AdminBundle/Resources/js/components/Popover/Popover.js

```diff
@@ -0,0 +1,148 @@
+
+type Props = {
+    open: boolean,
+    children: ChildrenArray<*>,
+    onClose?: () => void,
+    /** The element which will be used to position the popover */
+    anchorEl: ElementRef<*>,
+    /** 
+     * A child-node inside the popover which should be displayed on top of the anchor when the popover is open.
+     * Similar to the select element behaviour on OSX.
+     */
+    centerChildNode: ElementRef<*>,
+    /** The horizontal offset of the popover relative to the anchor element */
+    horizontalOffset: number,
+    /** The vertical offset of the popover relative to the anchor element */
+    verticalOffset: number,
+};
+
+@observer
+export default class Popover extends React.PureComponent<Props> {
+    static defaultProps = {
+        open: false,
+        horizontalOffset: 0,
+        verticalOffset: 0,
+    };
+
+    popoverEl: ElementRef<'div'>;
+
+    @observable scrollHeight: number;
+
+    @observable scrollWidth: number;
+
+    scrollTop: number;
+
+    componentDidMount() {
+        window.addEventListener('blur', this.close);
+        window.addEventListener('resize', this.close);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('blur', this.close);
+        window.removeEventListener('resize', this.close);
+    }
+
+    componentDidUpdate() {
+        if (this.popoverEl) {
+            this.popoverEl.scrollTop = this.scrollTop || 0;
+        }
+    }
+
+    close = () => {
+        if (this.props.onClose) {
+            this.props.onClose();
+        }
+    };
+
+    @computed get dimensions(): PopoverDimensions {
+        const {
+            anchorEl,
+            verticalOffset,
+            horizontalOffset,
+            centerChildNode,
+        } = this.props;
+        const {
+            top = 0,
+            left = 0,
+            width = 0,
+            height = 0,
+        } = anchorEl.getBoundingClientRect();
+        const centerChildOffsetTop = (centerChildNode) ? centerChildNode.offsetTop : 0;
+        const alignOnVerticalAnchorEdges = (centerChildNode) ? false : true;
+
+        return PopoverPositioner.getCroppedDimensions(
+            this.scrollWidth,
+            this.scrollHeight,
+            top,
+            left,
+            width,
+            height,
+            horizontalOffset,
+            verticalOffset,
+            centerChildOffsetTop,
+            alignOnVerticalAnchorEdges,
+        );
+    }
+
+    @action setScrollDimensions(scrollWidth: number, scrollHeight: number) {
+        this.scrollWidth = scrollWidth;
+        this.scrollHeight = scrollHeight;
+    }
+
+    handleBackropClick = this.close;
+
+    setPopoverNode = (popoverEl: ElementRef<'div'>) => {
+        if (!popoverEl) {
+            return;
+        }
+
+        this.popoverEl = popoverEl;
+
+        const {
+            scrollWidth,
+            scrollHeight,
+        } = popoverEl;
+        const borderWidth = parseInt(window.getComputedStyle(popoverEl).borderLeftWidth, 10) * 2;
```

Comment:
Why is the value here multiplied by two?

Reply:
Ah, get it... But it is probably not correct (what if the left and right borders are different?).

Apart from that, why do you need the border width here?